monai/transforms/composables.py

```diff
@@ -37,77 +38,130 @@
 @alias('SpacingD', 'SpacingDict')
 class Spacingd(MapTransform):
     """
-    dictionary-based wrapper of :py:class:`monai.transforms.transforms.Spacing`.
+    Dictionary-based wrapper of :py:class:`monai.transforms.transforms.Spacing`.
+
+    This transform assumes the ``data`` dictionary has a field for the input
+    data's affine.  The field is created by either ``meta_key_format.format(key,
+    'affine')`` or ``meta_key_format.format(key, 'original_affine')``.
+
+    After resampling the input array, this transform will write the affine
+    after resampling to the field ``meta_key_format.format(key, 'affine')``,
+    at the same time, if ``meta_key_format.format(key, 'original_affine')`` doesn't exist,
+    the field will be created and set to the affine before resampling.
+
+    if no affine is specified in the input data, defauting to "eye(4)".
+
+    see also:
+        :py:class:`monai.transforms.transforms.Spacing`
     """
 
-    def __init__(self, keys, affine_key, pixdim, interp_order=2, keep_shape=False, output_key='spacing'):
+    def __init__(self, keys, pixdim, diagonal=False, mode='constant', cval=0,
+                 interp_order=3, dtype=None, meta_key_format='{}.{}'):
         """
         Args:
-            affine_key (hashable): the key to the original affine.
-                The affine will be used to compute input data's pixdim.
             pixdim (sequence of floats): output voxel spacing.
+            diagonal (bool): whether to resample the input to have a diagonal affine matrix.
+                If True, the input data is resampled to the following affine::
```

Comment:
typo '::'

Reply:
that's not a typo, it's a literal block in restructuredtext https://docutils.sourceforge.io/docs/user/rst/quickref.html#literal-blocks